beacon-chain/db/filesystem/blob.go

```diff
@@ -250,22 +250,28 @@ func (bs *BlobStorage) Prune(currentSlot primitives.Slot) error {
 
 	log.Debug("Pruning old blobs")
 
-	folders, err := afero.ReadDir(bs.fs, ".")
-	if err != nil {
-		return err
-	}
 	var totalPruned int
-	for _, folder := range folders {
-		if folder.IsDir() {
-			num, err := bs.processFolder(folder, currentSlot, retentionSlots)
+	err = afero.Walk(bs.fs, ".", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			log.WithError(err).Debugf("Failed to walk path %s", path)
+			return nil
+		}
+		if info.IsDir() && path != "." {
+			num, err := bs.processFolder(info, currentSlot, retentionSlots)
 			if err != nil {
-				return err
+				log.WithError(err).Debugf("Failed to process folder %s", info.Name())
```

Comment:
Could we use `errors.Wrap` to bundle the name of the failed folder and return it? That would result in a more detailed `log.Error` in the call to `tryPrune` under pr #13355. I think that will also shortcircuit the `Walk`, so there's a tradeoff if we're worried about a bad dir/file causing walk to get stuck for all subsequent folders.

Reply:
Sounds good but let's merge #13355 first. Will have to test it. Becuase without this, my node died at start up because replay state fails

Reply:
I just went back to your PR description and saw that you addressed this point, but I've got a couple questions/comments.

`Prune` is called in a goroutine, where it's error result is logged at Error level. So returning an error in walk->prune doesn't not impact the Save call that triggered it. 

Still, I think it makes sense to keep walking rather than causing the whole prune to fail, but I think this should be error log level rather than debug. Maybe we should return the last error we receive, or return a separate error if any directories fail, so that the error log in prune is triggered, because it could be confusing that some error paths under Prune log with a particular message and others don't. 

I don't understand this:
> This approach is essential when replaying state, as replaying will attempt to prune the blob and may otherwise raise issues with any non-existent folder.

Why would replaying state trigger pruning?

Reply:
> Why would replaying state trigger pruning?


I may be wrong here, but I thought replaying state triggers `ReceiveBlock`

Reply:
If replaying state does trigger ReceiveBlock, that should still not result in blob saving.